src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs

```diff
@@ -173,47 +174,66 @@ public override int GetMatsOsPlatformCode()
+                    {
+                        FileName = url,
+                        UseShellExecute = true
+                    };
+                    Process.Start(psi);
+                }
+                catch
                 {
+                    // hack because of this: https://github.com/dotnet/corefx/issues/10361
                     url = url.Replace("&", "^&");
                     Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
                 }
-                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                try
                 {
-                    try
+                    ProcessStartInfo psi = null;
+                    foreach (string openTool in new[] { "xdg-open", "gnome-open", "kfmclient" })
                     {
-                        Process.Start("xdg-open", url);
+                        if (TryGetExecutablePath(openTool, out string openToolPath))
+                        {
+                            psi = new ProcessStartInfo(openToolPath, url)
+                            {
+                                RedirectStandardOutput = true,
+                                RedirectStandardError = true
+                            };
+
+                            Process.Start(psi);
+
+                            break;
+                        }
                     }
-                    catch (Exception ex)
+
+                    if (psi == null)
                     {
-                        throw new MsalClientException(
-                            MsalError.LinuxXdgOpen,
-                            "Unable to open a web page using xdg-open. See inner exception for details. Possible causes for this error are: xdg-open is not installed or " +
-                            "it cannot find a way to open an url - make sure you can open a web page by invoking from a terminal: xdg-open https://www.bing.com ",
-                            ex);
+                        throw new Exception("Failed to locate a utility to launch the default web browser.");
                     }
                 }
-                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                catch (Exception ex)
                 {
-                    Process.Start("open", url);
-                }
-                else
-                {
-                    throw new PlatformNotSupportedException(RuntimeInformation.OSDescription);
+                    throw new MsalClientException(
+                        MsalError.LinuxXdgOpen,
+                        MsalErrorMessage.LinuxOpenToolFailed,
+                        ex);
                 }
             }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                Process.Start("/usr/bin/open", url);
```

Comment:
@mjcheetham - is this what you meant ? Or to search for "open" using `TryGetExecutablePath` ?

Reply:
The framework decided to use a hardcoded string to `/usr/bin/open`, so I'd say it's OK to follow that.